### DISCUSSION TITLE
Streaming api shows 404 error when cloudflare proxy is enabled.

### DISCUSSION THREAD

**Discussion by dungsil** [mastodon/mastodon#20988] [Q&A] [answered]
In dns only mode, websocket is connected normally, but if set to proxy mode, 404 error is displayed. The streaming api log shows `Subscription error: Unknown stream type`.

dns only:
![화면 캡처 2022-11-18 160509](https://user-images.githubusercontent.com/39783731/202642758-48d642e9-f55f-4a04-be66-141d6370e394.png)

proxy mode:
![화면 캡처 2022-11-18 160842](https://user-images.githubusercontent.com/39783731/202642860-11fcd27a-c61a-4584-af4c-3ea925bb02a5.png)

Nginx config file: https://github.com/dun-land/mastodon/blob/main/config/nginx.conf

**Comment by dungsil** [comment, marked as answer]
My cloudflare config was rejecting websocket connections....

**Reply by NicolaeNMV** [reply]
Hi @dungsil , can you elaborate how did you fix it ? I'm also having trouble with cloudflare & mastodon..

**Reply by dungsil** [reply]
Hello @NicolaeNMV ,
Check the **Websocket** entry in the Cloudflare Dashboard > Your Domain > Network.

If correct, it would be:
![image](https://user-images.githubusercontent.com/39783731/203570976-95c14010-c05f-4b41-8d3e-f30088363690.png)
